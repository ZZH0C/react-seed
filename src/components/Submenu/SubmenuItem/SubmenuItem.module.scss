.reset_button:focus{
  border-left: 5px #646d81 !important;
  padding-left: 30px !important;
}