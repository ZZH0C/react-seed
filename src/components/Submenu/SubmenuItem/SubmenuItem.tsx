import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import { useQueryParams } from '../../../hooks/useQueryParams/useQueryParams';
import styles from './SubmenuItem.module.scss';
import classNames from 'classnames';

interface MainSectionSubmenuItemProps
  extends React.HTMLAttributes<HTMLElement> {
  name: string;
}

export const SubmenuItem: React.FC<MainSectionSubmenuItemProps> = ({
  name,
  children,
}) => {
  const { pathname } = useLocation();
  const { removeParam } = useQueryParams();
  const clearParams = removeParam('in');
  return (
    <section className="submenuSection">
      <h4 className="submenuCategory">{name}</h4>
      <ul className="nav">
        <li>
          <Link
            className={classNames(styles.reset_button)}
            to={{
              pathname,
              search: clearParams,
            }}
          >
            Reset Category
          </Link>
        </li>
        {children}
      </ul>
    </section>
  );
};
