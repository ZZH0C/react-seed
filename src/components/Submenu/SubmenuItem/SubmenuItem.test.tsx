import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { SubmenuItem } from './SubmenuItem';

describe('components/ButtonLink', () => {
  const mockProps = {
    name: 'SOME_NAME',
    children: 'SOME_TEXT',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(
      shallow(
        <SubmenuItem name={mockProps.name}>{mockProps.children}</SubmenuItem>,
      ),
    ).toMatchSnapshot();
  });
});
