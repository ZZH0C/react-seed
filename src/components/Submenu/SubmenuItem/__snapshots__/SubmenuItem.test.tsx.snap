// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonLink should render and match snapshot 1`] = `
<section
  className="submenuSection"
>
  <h4
    className="submenuCategory"
  >
    SOME_NAME
  </h4>
  <ul
    className="nav"
  >
    SOME_TEXT
  </ul>
</section>
`;
